engines/sci/graphics/palette32.cpp

```diff
@@ -21,20 +21,156 @@
  */
 
 #include "common/file.h"
+#include "common/system.h"
+#include "graphics/palette.h"
 
 #include "sci/sci.h"
+#include "sci/resource.h"
 #include "sci/graphics/palette32.h"
+#include "sci/graphics/screen.h"
 
 namespace Sci {
 	
 GfxPalette32::GfxPalette32(ResourceManager *resMan, GfxScreen *screen)
-	: GfxPalette(resMan, screen), _clutTable(0), _cyclers(), _cycleMap() {}
+	: GfxPalette(resMan, screen),
+	_clutTable(nullptr),
+	// Palette cycling
+	_cyclers(), _cycleMap(),
+	// Palette varying
+	_varyTime(0), _varyDirection(0), _varyTargetPercent(0),
+	_varyTargetPalette(nullptr), _varyStartPalette(nullptr),
+	_varyFromColor(0), _varyToColor(255), _varyNumTimesPaused(0),
+	_varyPercent(_varyTargetPercent), _varyLastTick(0),
+	// Palette versioning
+	_version(1), _needsUpdate(false) {
+		// TODO: Sci::Palette should have a default constructor that does
+		// all of this
+		for (int color = 0; color < ARRAYSIZE(_sysPalette.colors); color++) {
+			_sourcePalette.colors[color] = _nextPalette.colors[color] = _sysPalette.colors[color];
+			// TODO: Intensity and mapping are not used by SCI32 but are used by
+			// other parts of the graphic engine right now. Replace those parts.
```

Comment:
Sorry, I guess this was a really bad/confusing comment. Because GfxPalette32 inherits from GfxPalette16 there is possibly still some code that relies on intensity and mapping being set to these defaults (I don’t think this is true any more, but I won’t know for sure until I have audited everything). However, SCI32 SOLPalettes don’t have intensity or mapping information (this data is now stored primarily on PaletteMgr) and so the engine shouldn’t ever be using this information for SCI32 games. This was supposed to be a note to myself that any part of the graphics engine that is using intensity/mapping needs to be changed to *not* do that for SCI32 games.